caffe2/serialize/func_adapter.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <functional>
+#include "c10/macros/Macros.h"
+#include "caffe2/serialize/read_adapter_interface.h"
+
+namespace caffe2 {
+namespace serialize {
+
+// Read a number of bytes into a buffer and return the number of bytes actually
+// read
+using ReaderFunc = std::function<size_t(void*, size_t)>;
```

Comment:
`std::function` is notoriously slow. Do we know that it's use here is acceptable?